src/resonanceReconstruction/rmatrix/legacy/unresolved/Data.hpp

```diff
@@ -0,0 +1,52 @@
+/**
+ *  @class
+ *  @brief Generic data helper class for legacy unresolved data
+ */
+template < typename Quantity >
+struct Data {
+
+  Quantity elastic;
+  Quantity capture;
+  Quantity fission;
+  Quantity competition;
+
+  Data( Quantity elastic, Quantity capture,
+        Quantity fission, Quantity competition ) :
+    elastic( elastic ), capture( capture ),
+    fission( fission ), competition( competition ) {}
+
+  Data operator+( const Data& other ) const {
+    return { this->elastic + other.elastic,
+             this->capture + other.capture,
+             this->fission + other.fission,
+             this->competition + other.competition };
+  }
+
+  Data operator-( const Data& other ) const {
+    return { this->elastic - other.elastic,
+             this->capture - other.capture,
+             this->fission - other.fission,
+             this->competition - other.competition };
+  }
+
+  Data& operator+=( const Data& other ) {
+    this->elastic += other.elastic;
+    this->capture += other.capture;
+    this->fission += other.fission;
+    this->competition += other.competition;
+    return *this;
+  }
+
+  Data& operator-=( const Data& other ) {
+    this->elastic -= other.elastic;
+    this->capture -= other.capture;
+    this->fission -= other.fission;
+    this->competition -= other.competition;
+    return *this;
+  }
+
+  bool hasElastic() const { return this->elastic != Quantity(); }
+  bool hasCapture() const { return this->capture != Quantity(); }
+  bool hasFission() const { return this->fission != Quantity(); }
+  bool hasCompetition() const { return this->competition != Quantity(); }
```

Comment:
What is the value really is `0.0`? This would show that there isn't a value here. Perhaps a value of `0.0` is the same as not having it?

Reply:
These are used for widths, degrees of freedom, etc. In those cases, if it's zero, it's as if it isn't there. This has an effect on how the fluctuation integrals are calculated.